### PR TITLE
Fix failing data stream snapshot test

### PR DESCRIPTION
**Pull request by danhermann**
Fixes the test that failed in https://gradle-enterprise.elastic.co/s/7zkrlus7vgn2s

### CODE REVIEW

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/1

File: server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java [hunk 1]
```diff
@@ -205,7 +205,11 @@ public void testSnapshot() {
 
     public void testSnapshotWithAllBackingIndicesRemoved() {
         var preSnapshotDataStream = DataStreamTestHelper.randomInstance();
-        var indicesToAdd = DataStreamTestHelper.randomIndexInstances();
+        var indicesToAdd = new ArrayList<Index>();
```
Comment (original-brownbear) [inline server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java:208]: NIT: I suppose you can shorten this via 

```
randomValueOtherThan(Collections.emptyList(), DataStreamTestHelper::randomIndexInstances)
```

**Review by original-brownbear — APPROVED** [review]
LGTM

**Comment by danhermann** [comment]
Thanks, @original-brownbear!

**Pull request merged**